### PR TITLE
Mapsd: Test recovery from internet drop

### PR DESCRIPTION
**Pull request by jnewb1**
Test if mapsd can recover after an internet drop, by doing a MITM and dropping the request

related to: https://github.com/commaai/openpilot/issues/28900

### CODE REVIEW

File: selfdrive/navd/tests/test_map_renderer.py [hunk 1]
```diff
@@ -9,18 +14,73 @@
...
+
 class TestMapRenderer(unittest.TestCase):
+  server = MapBoxInternetDisabledServer()
+
   @classmethod
   def setUpClass(cls):
     assert "MAPBOX_TOKEN" in os.environ
+    cls.original_token = os.environ["MAPBOX_TOKEN"]
+    cls.server.start()
+    time.sleep(2) # wait for server to be fully started
```
Comment (adeebshihadeh) [inline selfdrive/navd/tests/test_map_renderer.py:79]: does this really take any time?

File: selfdrive/navd/tests/test_map_renderer.py [hunk 2]
```diff
@@ -29,43 +89,50 @@ def setUp(self):
 
     if os.path.exists(CACHE_PATH):
       os.remove(CACHE_PATH)
-
+  
   def tearDown(self):
     managed_processes['mapsd'].stop()
-
-  def _run_test(self, expect_valid):
+    time.sleep(2) # wait for server to be fully stopped
```
Comment (adeebshihadeh) [inline selfdrive/navd/tests/test_map_renderer.py:95]: this is a long time to sleep between tests. is this really necessary?

File: selfdrive/navd/tests/test_map_renderer.py [hunk 3]
```diff
@@ -9,18 +14,73 @@
...
+    if self.INTERNET_ACTIVE:
+      super().setup()
+  
+  def handle(self):
+    if self.INTERNET_ACTIVE:
+      super().handle()
+
+  def finish(self):
+    if self.INTERNET_ACTIVE:
+      super().finish()
```
Comment (adeebshihadeh) [inline selfdrive/navd/tests/test_map_renderer.py:41]: do we actually need to hook all of these?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: selfdrive/navd/tests/test_map_renderer.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/navd/tests/test_map_renderer.py]: seems like this loads and stops fast enough that it isn't a problem

File: selfdrive/navd/tests/test_map_renderer.py [hunk 4]
```diff
@@ -29,43 +89,50 @@ def setUp(self):
...
       assert self.sm.updated['mapRenderState'] == frame_expected, "renderer running at wrong frequency"
 
       if not frame_expected:
-
         continue
 
+      frames_since_test_start = self.sm['mapRenderState'].frameId - starting_frame_id
+
       # give a few frames to go valid
-      if expect_valid and not self.sm.valid['mapRenderState'] and not prev_valid and self.sm['mapRenderState'].frameId < 5:
```
[the remark this answers is not in the document]
Comment (jnewb1) [reply selfdrive/navd/tests/test_map_renderer.py:68]: I put back in the valid -> invalid check. also needed an invalid -> valid check for the recover test

File: selfdrive/navd/tests/test_map_renderer.py [hunk 5]
```diff
@@ -29,43 +89,50 @@ def setUp(self):
 
     if os.path.exists(CACHE_PATH):
       os.remove(CACHE_PATH)
-
+
```
[the remark this answers is not in the document]
Comment (jnewb1) [reply selfdrive/navd/tests/test_map_renderer.py]: done

File: selfdrive/navd/tests/test_map_renderer.py [hunk 6]
```diff
@@ -83,16 +150,41 @@ def _run_test(self, expect_valid):
...
 
+  def disable_internet(self):
+    MapBoxInternetDisabledRequestHandler.INTERNET_ACTIVE = False
+  
+  def enable_internet(self):
+    MapBoxInternetDisabledRequestHandler.INTERNET_ACTIVE = True
+
   def test_with_internet(self):
+    self.location = LOCATION1
+    self.enable_internet()
```
[the remark this answers is not in the document]
Comment (jnewb1) [reply selfdrive/navd/tests/test_map_renderer.py]: done

**Pull request merged**